src/codeConventions.md

```diff
@@ -4,8 +4,1192 @@
+<pre><code><span class="keyword">public</span> <span class="keyword">boolean</span> isReference() { <span class="keyword">return</span> true; }</code></pre>
+</div>
+</div>
+<div class="rightcol">
+<div class="box">
+<div class="boxheader donts">Don’ts</div>
+<pre><code><span class="keyword">public</span> <span class="keyword">boolean</span> getResult() { <span class="keyword">int</span> value = getValue(); <span class="keyword">return</span> value &lt; <span class="literal">0</span> ? <span class="literal">0</span> : value; }</code></pre>
+<pre><code><span class="keyword">for</span> (<span class="keyword">int</span> i = <span class="literal">0</span>; i < size; i++) { sum += data[i]; }</code></pre>
+</div>
+</div>
+<p style="clear: both"></p>
+</div>
+    
+### Indentation{#indentation}
+<ul class="conventions">
+    <li>Indentation level is four spaces.</li>
+    <li>Only space characters may be used for indentation. No tabs.</li>
+    <li>Empty lines must not be indented. (This is implied by the no trailing white space rule.)</li>
+    <li>`case` lines should be indented with four spaces, and statements within the case should be indented with another four spaces.</li>
+</ul>
+Refer to Section&nbsp;[Wrapping Lines](#wrapping-lines)&nbsp;for guidelines on how to indent continuation lines.
+        
+<div>
+<div class="leftcol">
+<div class="box">
+<div class="boxheader dos">Dos</div>
+<pre><code><span class="keyword">switch</span> (var) {
+    <span class="keyword">case</span> TWO:
+        setChoice(<span class="literal">"two"</span>);
+        <span class="keyword">break</span>;
+    <span class="keyword">case</span> THREE:
+        setChoice(<span class="literal">"three"</span>);
+        <span class="keyword">break</span>;
+    <span class="keyword">default</span>:
+        <span class="keyword">throw</span> <span class="keyword">new</span> IllegalArgumentException();
+}</code></pre>
+</div>
+</div>
+<div class="rightcol">
+<div class="box">
+<div class="boxheader donts">Don’ts</div>
+<pre><code><span class="keyword">switch</span> (var) {
+<span class="keyword">case</span> TWO:
+    setChoice(<span class="literal">"two"</span>);
+    <span class="keyword">break</span>;
+<span class="keyword">case</span> THREE:
+    setChoice(<span class="literal">"three"</span>);
+    <span class="keyword">break</span>;
+<span class="keyword">default</span>:
+    <span class="keyword">throw</span> <span class="keyword">new</span> IllegalArgumentException();
+}</code></pre>
+</div>
+</div>
+<p style="clear: both"></p>
+</div>
+    
+<div class="box">
+<div class="boxheader">Motivation</div>
+By using spaces for indentation, problems regarding different default settings for tab stops in various editors are avoided.
+</div>
+
+### Wrapping Lines{#wrapping-lines}
+<ul class="conventions">
+<li>Source code and comments should generally not exceed 80 characters per line and rarely if ever exceed 100 characters per line, including indentation.
```

Comment:
Was this actually agreed to? 80 cars is a truly ancient number. Especially Java code tends to be slightly chatty and have longer lines, especially with a few levels of indentation. I wouldn't want to have lines up to 180-200 characters, but 120-140 is not excessive in my mind.

I'm looking at some Java code I wrote not long ago with an inline Thread - run() declaration inside an if, with a loop and a few ifs in it... This may be exceptionally bad Java code, but it has four spaces of indentation and 9 indentation steps in the deepest end. That's almost half of the 80 chars in just indentation. I personally don't find this code the least difficult to read. But then again, the actual lines themselves are not more than ~100 characters long, they are just moved to the right a bit due to indentation.  Could the 80-100 lines be excluding indentation?

Reply:
I think most people agree with you on this (including me). Questions like these however tend to be a bit contentious so we opted to just let the [existing conventions](https://www.oracle.com/java/technologies/javase/codeconventions-indentation.html#313) carry over here. This is probably something that's worth starting a separate discussion about, but I suggest we do that in a separate change.

Does this sound good to you?